xrootd/client/client.go

```diff
@@ -24,30 +24,30 @@ package client // import "go-hep.org/x/hep/xrootd/client"
 
 import (
 	"context"
-	"io"
-	"net"
-	"sync/atomic"
+	"sync"
 
-	"go-hep.org/x/hep/xrootd/internal/mux"
-	"go-hep.org/x/hep/xrootd/internal/xrdenc"
+	"github.com/pkg/errors"
 	"go-hep.org/x/hep/xrootd/xrdproto"
 	"go-hep.org/x/hep/xrootd/xrdproto/auth"
 	"go-hep.org/x/hep/xrootd/xrdproto/auth/krb5"
 	"go-hep.org/x/hep/xrootd/xrdproto/auth/unix"
-	"go-hep.org/x/hep/xrootd/xrdproto/sigver"
 )
 
 // A Client to xrootd server which allows to send requests and receive responses.
 // Concurrent requests are supported.
 // Zero value is invalid, Client should be instantiated using NewClient.
 type Client struct {
-	cancel           context.CancelFunc
-	conn             net.Conn
-	mux              *mux.Mux
-	protocolVersion  int32
-	signRequirements xrdproto.SignRequirements
-	seqID            int64
-	auths            map[string]auth.Auther
+	cancel   context.CancelFunc
+	auths    map[string]auth.Auther
+	username string
+	// initialSessionID is the sessionID of the server which is used as default
+	// for all requests that don't specify sessionID explicitly.
+	// Any failed request with another sessionID should be redirected to the initialSessionID.
+	// See http://xrootd.org/doc/dev45/XRdv310.pdf, page 11 for details.
+	initialSessionID string
+	mu               sync.RWMutex
+	sessions         map[string]*session
+	maxRedirections  int
```

Comment:
as `maxRedirections` isn't meant to be protected by the `sync.RWMutex`, please add an empty line before `maxRedirection` to clearly demark a logical block.